### PR TITLE
Update documentation to new format and updated github workflow

### PR DESCRIPTION
**Pull request by kapilraajP**
## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,269 +1,44 @@
...
+[![Build](https://github.com/ballerina-platform/module-ballerinax-twitter/workflows/CI/badge.svg)](https://github.com/ballerina-platform/module-ballerinax-twitter/actions?query=workflow%3ACI)
+[![GitHub Last Commit](https://img.shields.io/github/last-commit/ballerina-platform/module-ballerinax-twitter.svg)](https://github.com/ballerina-platform/module-ballerinax-twitter/commits/master)
+[![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)
 
-## What is Twitter?
-Twitter is a social networking service on which users can post and communicate with status called as "Tweets". Authenticated users can post, like, and retweet tweets.  Main use case is to connect people and allow people to express their thoughts to large set of people.
+[Twitter](https://twitter.com/) is a social media service provided by Twitter.
```
Comment (shaniR) [inline README.md:7]: Shall we use a rephrased version of the previous description? Twitter is a social networking service on which users can post and communicate with a status called "Tweets".  Authenticated users can post, like, and retweet tweets. The main goal of Twitter is to connect people and to provide a platform for every person to express their thoughts to a large set of people freely.

File: README.md [hunk 2]
```diff
@@ -1,269 +1,44 @@
...
+### Setting up the prerequisites
 
-## Supported Versions
+1. Download and install Java SE Development Kit (JDK) version 11. You can install either [OpenJDK](https://adoptopenjdk.net/) or [Oracle JDK](https://www.oracle.com/java/technologies/javase-jdk11-downloads.html).
 
-|                                   | Version               |
-|:---------------------------------:|:---------------------:|
-|       Twitter API Version         |         v1.1          |
-|       Ballerina Language          |   Swan Lake Alpha 5   |
+   > **Note:** Set the JAVA_HOME environment variable to the path name of the directory into which you installed JDK.
```
Comment (shaniR) [inline README.md:18]: Use markdown syntax for note, if cannot use this as a step.

File: README.md [hunk 3]
```diff
@@ -1,269 +1,44 @@
...
 
-## Limitations
+2. Download and install [Ballerina Swan Lake Beta2](https://ballerina.io/). 
 
-The connector doesn't support all the operations offered by Twitter API V1.1.
+### Building the source
+Execute the following commands to build from the source:
```
Comment (shaniR) [inline README.md:23]: The end should be a period and not a colon.

File: twitter/Module.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
+## Overview
+Connects to Twitter using Ballerina.
```
Comment (shaniR) [inline twitter/Module.md:2]: via [Ballerina](https://ballerina.io/).

File: twitter/Module.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+The Twitter API can be used to programmatically retrieve data and engage with the conversation on Twitter. When accessing Twitter through Ballerina Twitter Connector, It is required to register a developer application. By default, applications can only access public information on Twitter. Certain operations, such as those responsible for sending or receiving direct Messages and posting a tweet, require additional permissions from user before access user information. These permissions are not granted by default; you choose on a per-application basis whether to provide this access, and can control all the applications authorized on your account.
```
Comment (shaniR) [inline twitter/Module.md:4]: Let's rephrase this as: "The Twitter API can be used to programmatically retrieve data and to engage in a conversation on Twitter. When accessing Twitter through Ballerina Twitter Connector, it is required to register a developer application. By default, applications can only access public information on Twitter. Certain operations, such as those responsible for sending or receiving direct messages and posting a Tweet, require additional permissions from the user before accessing user information. These permissions are not granted by default, and you can choose on a per-application basis if to provide access. Furthermore, you can control all the applications authorized on your account."

File: twitter/Module.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+This module supports [Twitter API](https://developer.twitter.com/en/docs/twitter-api/v1) v1.1 version.
+
+## Prerequisites
+Before using this connector in your Ballerina application, complete the following:
+* Create [Twitter Developer Account](https://developer.twitter.com/en/apply-for-access)
```
Comment (shaniR) [inline twitter/Module.md:10]: Create a

File: twitter/Module.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+* Obtaining tokens
```
Comment (shaniR) [inline twitter/Module.md:11]: Obtain tokens

File: twitter/Module.md [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+        
+    Follow [this link](https://developer.twitter.com/en/docs/authentication/oauth-1-0a) and obtain the API key, API key secret, Access token and Access token secret.
```
Comment (shaniR) [inline twitter/Module.md:13]: obtain the API key, API key secret, Access token, and Access token secret

File: twitter/Module.md [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+## Quickstart
+
+To use the Twitter connector in your Ballerina application, update the .bal file as follows:
+
+### Step 1: Import connector
+First, import the ballerinax/twitter module into the Ballerina project.
```
Comment (shaniR) [inline twitter/Module.md:20]: Let's remove the word "First,"

File: twitter/Module.md [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+    apiSecret: "<apiSecret>",
+    accessToken: "<accessToken>",
+    accessTokenSecret: "<accessTokenSecret>"
+};
+
+twitter:Client twitterClient = check new(twitterConfig);
+```
+
+### Step 3: Invoke connector operation
+1. You can post a tweet as follows with `tweet` method by passing content as a parameter. Successful creation returns the created `Tweet` and the error cases returns an `error` object.
```
Comment (shaniR) [inline twitter/Module.md:37]: Let's remove descriptions as discussed.

File: twitter/Package.md [hunk 11]
```diff
@@ -1,214 +1,21 @@
...
-# Introduction
-The Twitter API can be used to programmatically retrieve and analyze data, as well as engage with the conversation on Twitter.
-This connector provides access to a variety of different resources in Twitter including the following:
-* Tweets
-* Users
- 
-The Twitter API currently consists of two supported versions, as well as different access tiers.
-When user wants to access Twitter through Ballerina Twitter Connector, they are required to register an application. By default, applications can only access public information on Twitter. Certain operations, such as those responsible for sending or receiving Direct Messages and posting a Tweet, require additional permissions from user before access user information. These permissions are not granted by default; you choose on a per-application basis whether to provide this access, and can control all the applications authorized on your account.
+## Package overview
+The `ballerinax/twitter` is a [Ballerina](https://ballerina.io/) connector for Twitter API.
```
Comment (shaniR) [inline twitter/Package.md:4]: Shall we use the connector name?

**Review by shaniR — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (kapilraajP) [reply README.md]: Fixed

File: README.md (hunk 3 above)
Comment (kapilraajP) [reply README.md]: Fixed

File: twitter/Module.md (hunk 4 above)
Comment (kapilraajP) [reply twitter/Module.md]: Fixed

File: twitter/Module.md (hunk 7 above)
Comment (kapilraajP) [reply twitter/Module.md]: Fixed

File: twitter/Module.md (hunk 10 above)
Comment (kapilraajP) [reply twitter/Module.md:37]: Fixed

File: twitter/Package.md (hunk 11 above)
Comment (kapilraajP) [reply twitter/Package.md:4]: Fixed

File: twitter/Module.md (hunk 9 above)
Comment (kapilraajP) [reply twitter/Module.md:20]: Fixed

File: README.md (hunk 2 above)
Comment (kapilraajP) [reply README.md:18]: This is how it is across all the connectors

**Review by abeykoon — APPROVED** [review]

File: twitter/endpoint.bal [hunk 12]
```diff
@@ -15,25 +15,33 @@
 // under the License.
 
 import ballerina/http;
+import ballerina/time;
 import ballerina/url;
+import ballerina/uuid;
 
-# Twitter Client object.
+# Ballerina Twitter connector provides the capability to access Twitter API.
+# This connector lets you to perform operations related to Tweets and users.
```
Comment (shaniR) [inline twitter/endpoint.bal:23]: lets you perform

File: twitter/endpoint.bal [hunk 13]
```diff
@@ -15,25 +15,33 @@
...
-    string accessTokenSecret;
+    private final string apiKey;
+    private final string apiSecret;
+    private final string accessToken;
+    private final string accessTokenSecret;
 
-    http:Client twitterClient;
+    private final http:Client twitterClient;
 
+    # Initializes the connector. During initialization you have to pass API credentials.
```
Comment (shaniR) [inline twitter/endpoint.bal:40]: To initialize, you have to pass API credentials.

File: twitter/endpoint.bal (hunk 12 above)
Comment (kapilraajP) [reply twitter/endpoint.bal:23]: Fixed

File: twitter/endpoint.bal (hunk 13 above)
Comment (kapilraajP) [reply twitter/endpoint.bal:40]: Corrected doc

File: twitter/Module.md (hunk 5 above)
Comment (kapilraajP) [reply twitter/Module.md:4]: Added as mentioned

File: twitter/Module.md (hunk 6 above)
Comment (kapilraajP) [reply twitter/Module.md:10]: Ack. Changed
